### DISCUSSION TITLE
how to finetune stable diffusion img2img model with only one input channel or gray scale image?

### DISCUSSION THREAD

**Discussion by sapkun** [huggingface/diffusers#6782] [General]
Hi! experts, I want to finetune stable diffusion img2img model (instructpix2pix or controlnet) with only one input channel or gray scale image. is that possible to do that or is there any example code? 
Thanks in advance.

**Comment by sapkun** [comment]
@sayakpaul Thanks for your reply.

**Comment by sapkun** [comment]
@patrickvonplaten @sayakpaul 
Looking forward to your answer! 
Thanks in advance!

**Comment by sayakpaul** [comment]
What happens if you just reduce the number of channels at initialization and prepare your dataset accordingly to meet the needs? 

We don't have any example code snippet for this but I believe the existing code snippets should be useful references already.

**Reply by sapkun** [reply]
Thank you for your reply. 
As for [instruct_pix2pix,](https://github.com/huggingface/diffusers/blob/main/examples/instruct_pix2pix/train_instruct_pix2pix.py) i found in here https://github.com/huggingface/diffusers/blob/02338c93178461f6dfbc9e82f352d03277812f82/examples/instruct_pix2pix/train_instruct_pix2pix.py#L460 i can change the in_channel, but as for [controlnet](https://github.com/huggingface/diffusers/blob/main/examples/controlnet/train_controlnet.py) or [t2i_adapter,](https://github.com/huggingface/diffusers/blob/main/examples/t2i_adapter/train_t2i_adapter_sdxl.py) i didn't find anywhere to change. i wonder is that feasible to reduce the number of in_channel [controlnet](https://github.com/huggingface/diffusers/blob/main/examples/controlnet/train_controlnet.py) and [t2i_adapter?](https://github.com/huggingface/diffusers/blob/main/examples/t2i_adapter/train_t2i_adapter_sdxl.py)
